### ISSUE TITLE
Disabling BracesAroundHashParameters for select methods

### ISSUE THREAD

**Issue by kranthiu** [rubocop-hq/rubocop#4570]
Methods like `render` `has_many` `enum` etc take a single hash parameter. Including {} will cause ruby to interpret the argument as a block instead. The  only way out is

    enum({status: { active: 0, archived: 1 }})

Is there any alternative ? Can we disable the rule for some specific methods ?

**Comment by Drenmi** [comment]
Hm. I think this cop should not recommend adding braces at all (for any method), in the case of an unparenthesized hash argument, as it changes the behaviour of the code. 🙂

**Comment by kranthiu** [comment]
We want disable braces only for the methods like `render` `has_many` `enum` etc. For the rest, we want to enforce braces.

We are using `EnforcedStyle: braces` but rubocop is (rightly) flagging `enum status: {active: 0, archived: 1}`  Does this cop support `IgnorePattern` or something similar ?

**Comment by rlue** [comment]
@kranthiu, can you clarify what the problem is? The convention in Rails is to leave out both the parentheses and the outer braces:

```ruby
enum status: { active: 0, archived: 1 }
```

As @Drenmi explains, this is acceptable according to the `BracesAroundHashParameters` cop, and is does not result in Ruby interpreting anything as a block.

---

On a separate note, @Drenmi, can you clarify what you mean here?

> adding braces... in the case of an unparenthesized hash argument... changes the behaviour of the code. 

I just submitted another issue (#5212) where my code breaks without the braces, but the cop is flagging them as redundant. The code seems to be doing what I intend; how would the braces "change its behavior" in this case?

**Comment by Drenmi** [comment]
It's slightly different @rlue, since your example has more than one argument.

Compare to a single argument:

```
$ ruby-parse -e 'foo bar: baz'
(send nil :foo
  (hash
    (pair
      (sym :bar)
      (send nil :baz))))
```

```
$ ruby-parse -e 'foo { bar: baz }'
(fragment:0):1:10: error: unexpected token tCOLON
(fragment:0):1: foo { bar: baz }
(fragment:0):1:
```

**Comment by mikegee** [comment]
I think @rlue's comment here resolved this issue. The very similar bug in #5212 remains open, though.

**Comment by bbatsov** [comment]
We're in the process of moving all Rails-related functionality to a standalone library (gem) named `rubocop-rails`. Please, migrate this issue to its issue tracker https://github.com/rubocop-hq/rubocop-rails/issues/
